gke-deploy/core/image/image.go

```diff
@@ -47,31 +40,32 @@ func ParseImages(ctx context.Context, images []string) ([]Image, error) {
 			return nil, fmt.Errorf("duplicate image name: %q", imName)
 		}
 		exists[imName] = true
-		ims = append(ims, im)
+		refs = append(refs, ref)
 	}
 
-	return ims, nil
+	return refs, nil
 }
 
-func parseImage(ctx context.Context, image string) (Image, error) {
+func parseImage(image string) (name.Reference, error) {
 	im, err := name.ParseReference(image)
 	if err != nil {
 		return nil, fmt.Errorf("image is invalid: %q", image)
 	}
 	return im, nil
 }
 
-// GetNameFromString gets an image's name, given a string representation of the image.
-func GetNameFromString(ctx context.Context, image string) (string, error) {
-	im, err := parseImage(ctx, image)
+// ParseImageReference gets an image's name, given a string representation of the image.
+// e.g., given "gcr.io/my-project/my-app:1.0.0", returns "gcr.io/project/my-app"
+func ParseImageReference(image string) (string, error) {
+	im, err := parseImage(image)
 	if err != nil {
 		return "", err
 	}
-	return GetName(ctx, im)
+	return GetName(im)
 }
 
 // GetName gets an image's name.
-func GetName(ctx context.Context, image Image) (string, error) {
+func GetName(image name.Reference) (string, error) {
 	switch t := image.(type) {
```

Comment:
`name.Reference` has all the methods you need, and you don't actually need to care whether it's a `name.Tag` or a `name.Digest` (or a future type that doesn't exist yet). Just `return fmt.Sprintf("%s/%s", ref.RegistryStr(), ref.RepositoryStr())`, and at this point `GetName` doesn't even need to return an error.

Reply:
https://github.com/google/go-containerregistry/blob/master/pkg/name/ref.go#L24
name.Reference doesn't have RegistryStr and RepositoryStr methods.

Reply:
Sorry, you're right, `ref.Context().RegistryStr()` and `ref.Context().RepositoryStr()`

https://play.golang.org/p/Da8RxUZTlvL

Reply:
If name.Reference has RegistryStr and RepositoryStr methods I can remove the GetName method

Reply:
^ Sweet! I'll get to refactoring